File: Backend/C#/SuperHero.Repository/Persister/BaseHeroPersister.cs
using SuperHero.Domain.Model;
using SuperHero.Repository.Behavior;
using SuperHero.Repository.Context;

namespace SuperHero.Repository.Persister
{
    public class BaseHeroPersister : IBaseHeroPersister
    {
        private readonly HeroContext _context;

        public BaseHeroPersister(HeroContext context)
        {
            _context = context;
        }

        public async Task<bool> CreateBaseHero(IEnumerable<BaseHero> baseHeroCollection)
        {
            await _context.AddRangeAsync(baseHeroCollection);
            await _context.SaveChangesAsync();

            return true;
        }
    }
}

File: Backend/Typescript/src/repository/persister/base-hero.persister.ts
import { inject, injectable } from "tsyringe";
import type { IBaseHeroPersister } from "../../domain/behavior/repository/base-hero-persister.interface.js";
import { BaseHero } from "../../domain/entities/base/base-hero.js";
import type { PrismaClient } from "@prisma/client";
import { baseHeroToPrisma } from "../mappers/mapToPrisma.js";

@injectable()
export class BaseHeroPersister implements IBaseHeroPersister {
    constructor(@inject("PrismaClient") private prisma: PrismaClient) { }
    async createBaseHero(baseHeroCollection: BaseHero[]): Promise<boolean> {
        try {
            for (const hero of baseHeroCollection) {
                await this.prisma.baseHero.upsert({
                    where: { privateId: hero.PrivateId! },
                    update: {},
                    create: baseHeroToPrisma(hero),
                });
            }
            console.log('Persisting base heroes to the database...');
            return true;
        } catch (error) {
            console.error('Error persisting base heroes:', error);
            throw error;
        }
    }
}